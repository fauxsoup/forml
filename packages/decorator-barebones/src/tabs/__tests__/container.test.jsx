import Container from '../container';
import Tab from '../tab';
import Panel from '../panel';
import React from 'react';
import { render } from '@testing-library/react';

describe('renders', function () {
    let form;
    let tabForm;
    let panelForm;
    beforeEach(function () {
        form = { type: 'tabs', tabs: [{ title: 'test' }] };
        tabForm = { title: 'tab title' };
        panelForm = tabForm;
    });
    test('its tabs and panels', function () {
        const { container } = render(
            <Container
                tabs={[<Tab form={tabForm} />]}
                panels={[
                    <Panel form={panelForm}>
                        <div id="panel_test">test</div>
                    </Panel>,
                ]}
            />
        );

        expect(container).toMatchSnapshot();
        expect(container.querySelector('button')).not.toBeNull();
        expect(container.querySelector('button').textContent).toBe('tab title');
        expect(container.querySelector('#panel_test')).not.toBeNull();
    });
});
