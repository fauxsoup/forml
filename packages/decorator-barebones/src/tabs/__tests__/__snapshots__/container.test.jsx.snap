// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders its tabs and panels 1`] = `
<div>
  <div
    class="tabs-container"
  >
    <div
      class="tabs"
    >
      <button>
        tab title
      </button>
    </div>
    <div
      class="panels"
    >
      <div>
        <div
          id="panel_test"
        >
          test
        </div>
      </div>
    </div>
  </div>
</div>
`;
